frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -124,6 +125,18 @@ def ensure_valid_state
       end
     end
 
+    def ensure_store_credits_eligible
+      expired_payments = @order.payments.valid.store_credits.select { |p| p.source.expired? }
```

Comment:
> Yet this behaviour is absolutely essential for store credits to work now.

I don't entirely agree with this. For users with custom frontends, not including this change will simply maintain current behaviour. If they wish to use expiring credits, they will have to pull this into their frontend, similar to any of the other logic that happens in `before_action`s in the `CheckoutController`.

> I would expect the exact opposite of that behaviour to be desired. Admins shouldn't accidentally be able to use expired store credits.

In our store - and presumably others - there are cases where the order may be held for review before completion (ie: fraud review). For cases like this, we can't guarantee that the order state at the time of checkout will be the same as when it gets approved. It's possible that the credits were valid when the user checked out, but have expired by the time an admin has a chance to review the order.

Another very common customer service use case is users that make it to the confirm page and think they completed the order. When they realize they haven't gotten an order confirmation email, they call customer service to have them complete the order and we can run into the same issue.

There is some potential for admins to accidentally use expired credit, but I think that's better than putting this in a transition hook that prevents them from completing these orders altogether, when there are valid use cases for doing so.